content/en/guide/v10/web-components.md

```diff
@@ -50,6 +50,12 @@ function Foo() {
 }
 ```
 
+> [!Note]
+> Preact maps only camelCase JSX property names to the same camelCase JS properties of the custom elements,
+> and does not map dash-case JSX properties to camelCase JS properties like other JSX compilers or runtimes might.
+> If a custom element has a property named `someProperty`, using the syntax `someProperty={value}` will work while
+> `some-property={value}` will not (the latter will set the attribute instead).
```

Comment:
```suggestion
> **Note:** Preact makes no assumptions over naming schemes and will not attempt to coerce names, in JSX or otherwise, to DOM properties. If a custom element has a property name `someProperty`, it will need to be set using `someProperty=...` rather than `some-property=...`.
```

Slightly different note syntax that I think our docs, by and large anyhow, prefers.

I'd skip over the bits about other frameworks or compilers, it's not particularly useful information. We'll also skip over the "will set the attribute instead" comment as that's not guaranteed, depending on the circumstances (albeit, it'd be an oddity).

Reply:
>  We'll also skip over the "will set the attribute instead" comment as that's not guaranteed, depending on the circumstances (albeit, it'd be an oddity).

That's curious. Do you mean, for example, if the element does have a `foo-bar` JS property then that won't set the attribute? Or is there another case?

Reply:
Precisely, yes.

The logic largely boils down to:

```js
if (!somethingThatNeedsToBeSetAsAnAttr && prop in domElement) domElement.prop = ...
else domElement.setAttribute(prop, ...)
```

So if `domElement['some-property']` exists, Preact will happily update the property, rather than falling through to the attribute.